zjit/src/codegen.rs

```diff
@@ -630,6 +630,26 @@ fn gen_setglobal(jit: &mut JITState, asm: &mut Assembler, id: ID, val: Opnd, sta
 
 /// Side-exit into the interpreter
 fn gen_side_exit(jit: &mut JITState, asm: &mut Assembler, reason: &SideExitReason, state: &FrameState) {
+    if get_option!(stats) {
+        let counter = match reason {
+            SideExitReason::UnknownNewarraySend(_) => Counter::specific_exit_unknown_newarray_send,
+            SideExitReason::UnknownCallType => Counter::specific_exit_unknown_call_type,
+            SideExitReason::UnknownOpcode(_) => Counter::specific_exit_unknown_opcode,
+            SideExitReason::UnhandledInstruction(_) => Counter::specific_exit_unhandled_instruction,
+            SideExitReason::FixnumAddOverflow => Counter::specific_exit_fixnum_add_overflow,
+            SideExitReason::FixnumSubOverflow => Counter::specific_exit_fixnum_sub_overflow,
+            SideExitReason::FixnumMultOverflow => Counter::specific_exit_fixnum_mult_overflow,
+            SideExitReason::GuardType(_) => Counter::specific_exit_guard_type_failure,
+            SideExitReason::GuardBitEquals(_) => Counter::specific_exit_guard_bit_equals_failure,
+            SideExitReason::PatchPoint(_) => Counter::specific_exit_patchpoint,
+            SideExitReason::CalleeSideExit => Counter::specific_exit_callee_side_exit,
+            SideExitReason::ObjToStringFallback => Counter::specific_exit_obj_to_string_fallback,
+            SideExitReason::UnknownSpecialVariable(_) => Counter::specific_exit_unknown_special_variable,
+            SideExitReason::UnhandledDefinedType(_) => Counter::specific_exit_unhandled_defined_type,
+            SideExitReason::Interrupt => Counter::specific_exit_interrupt,
+        };
+        gen_incr_counter(asm, counter);
+    }
```

Comment:
This only covers side exits created by `Insn::SideExit`, but other instructions generate side exits too. Let's move this thing to `Assembler::compile_side_exits`.